x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/TextEmbedding/TextEmbeddingServiceTests.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ *
+ * This file was contributed to by a Generative AI
+ */
+
+package org.elasticsearch.xpack.inference.services.TextEmbedding;
+
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.inference.InferenceServiceExtension;
+import org.elasticsearch.inference.Model;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.inference.services.settings.MlNodeDeployedServiceSettings;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+import static org.mockito.Mockito.mock;
+
+public class TextEmbeddingServiceTests extends ESTestCase {
```

Comment:
Might be worth adding some tests to ensure that after parsing the request config, if it is not empty it throws.